athena-scripts/raw_stage_optimisation_solution/scripts/core_preprocessing_functions.py

```diff
@@ -167,15 +167,26 @@ def generate_raw_select_filter(json_data, database, table, filter_processed_dt):
             raise ValueError("filter value for event_processing_testing_criteria is not found within config rules")
         print(f'config rule: event_processing_testing_criteria | filter: {event_processing_testing_criteria_filter}')
 
-        sql=f'''select * from \"{database}\".\"{table}\"'''
-
+        if stage_table_exists:
+            sql=f'''select * from \"{database}\".\"{table}\" where_clause'''
+        else:  
+            subquery = f'''select *,
+			                row_number() over (
+				            partition by event_id
+				            order by cast(concat(year, month, day) as int) desc
+			            ) as row_num
+               		    from \"{database}\".\"{table}\" as t
+                        where_clause'''
+            sql=f'''select * from ({subquery}) where row_num = 1'''
```

Comment:
this `sql` does not have 'where_clause' like the other one does for the string replaces below to use is that ok?